File: mandelbrot.c
// Mandelbrot Fun! - Dennis@VR
// z(n+1) = z(n+1)^2 + c
// OR z = z*z + c

#include <stdlib.h>
#include <stdio.h>
#include <GL/glut.h>
#include <time.h>

void main(int argc, char** argv)
{
	int mwidth = 800; 
	int mheight = 600;
	
	//Initialize GL and set Window
	glutInit(&argc, argv);
    glutInitDisplayMode(GLUT_SINGLE | GLUT_RGB );
    glutInitWindowSize(mwidth, mheight);
	glutCreateWindow("Mandelbrot");		
    glutReshapeWindow(mwidth, mheight);
	
	//Set GL Coordinates to 0,0 - Pixel mode
	glViewport(0, 0, mwidth, mheight);
	glMatrixMode(GL_PROJECTION);
	glLoadIdentity();
	glOrtho(0, mwidth, mheight, 0, -1, 1);
	glMatrixMode(GL_MODELVIEW);
	glLoadIdentity();
		
	glClear(GL_COLOR_BUFFER_BIT);
    glPointSize(1); 	
	
	short int h;
	int hx, hy;
	double real,imaginary;
	double z, zx, zy;

	clock_t t1 = clock(); 	
    glBegin(GL_POINTS); 	

	for (hx=1; hx<=mwidth; hx++)
	{
		real = (float)hx / 300 - 2.0;

		for (hy=1; hy<=mheight; hy++)  
			{
			imaginary = (float)hy / 300 - 1.0;
			zx = 0.0; 
			zy = 0.0;
			
			for (h=1; h<25; h++)  
			{
				//(x + yi) (x + yi) = (xx - yy) + (xy + xy)i. Complex Numbers by hand.
				z  = zx*zx - zy*zy + real;
				zy = zx*zy + zx*zy + imaginary;
				zx = z;
				
				if ( zx*zx + zy*zy > 4)  
				{
				glColor3b(h*6, h*3, h*1);
				glVertex2i(hx, hy);
				break;
				}
								
			}

		}
	}
	
	glEnd();
	
	glFlush();
	clock_t t2 = clock(); 
	float mtime = ( ( (float)t2 - (float)t1 ) / CLOCKS_PER_SEC ) * 1000;   
	printf("%f milliseconds\n", mtime);   
	glutMainLoop();  

}

File: mandelbrot.py
# Mandelbrot Fun! - Dennis@VR
# z(n+1) = z(n+1)^2 + c
# OR z = z*z + c

from Tkinter import *
from datetime import datetime

mwidth = 800
mheight = 600

t = Tk()

pic = PhotoImage(width=mwidth,height=mheight)
pic.put('#000000',(mwidth-1,mheight-1)) 

canv = Canvas(t, width=mwidth, height=mheight)
canv.pack()

mtime = datetime.now()

for x in range(0,mwidth):
    real = x / 300.0 - 2.0
    
    for y in range(0,mheight):
        imaginary = y / 300.0 - 1.0

        c = complex(real, imaginary)
        z = 0

        for h in range(0,25):
            z = z*z + c

            if abs(z) > 2:
                pic.put('#%02x%02x%02x' % (h*10, h*5, h*2),(x,y))
                break
            

canv.create_image(0, 0, image=pic, anchor=NW)
print datetime.now() - mtime
t.mainloop()

